src/Microsoft.AspNet.Mvc.Core/Filters/DefaultFilterProvider.cs

```diff
@@ -31,6 +34,10 @@ public void OnProvidersExecuting([NotNull] FilterProviderContext context)
                 {
                     ProvideFilter(context, item);
                 }
+                context.Results =
+                    context.Results
+                    .OrderBy(filter => filter.Descriptor, FilterDescriptorCorsComparer.Comparer)
```

Comment:
Well, it does seems odd, but this is the only place which can guarantee that the cors filter is the first ```AuthorizationFilter``` to execute. The cleaner alternative would be to have a separate filter stage but its kind of an overkill. We had discussed this at length during design review and we chose to do it this way.